File: 백준/Silver/1929. 소수 구하기/소수 구하기.cc
#include <iostream>
#include <vector>
using namespace std;

void sieve_of_eratosthenes(int M, int N) {
    // 0, 1
    vector<bool> is_prime(N + 1, true);
    is_prime[0] = is_prime[1] = false;

    for (int i = 2; i * i <= N; i++) {
        // i배수는 소수 x
        if (is_prime[i]) {
            for (int j = i * i; j <= N; j += i) {
                is_prime[j] = false;
            }
        }
    }

    for (int i = M; i <= N; i++) {
        if (is_prime[i]) {
            cout << i << '\n'; // 소수 출력
        }
    }
}

int main() {
    int M, N;
    cin >> M >> N;
    sieve_of_eratosthenes(M, N);
    return 0;
}

File: 백준/Silver/1929. 소수 구하기/소수 구하기.py
def sieve_of_eratosthenes(M, N):
    is_prime = [True] * (N + 1)
    is_prime[0] = is_prime[1] = False  # 0, 1

    for i in range(2, int(N**0.5) + 1):
        if is_prime[i]:
            for j in range(i * i, N + 1, i):
                is_prime[j] = False  # i의 배수

    for i in range(M, N + 1):
        if is_prime[i]:
            print(i)

M, N = map(int, input().split())
sieve_of_eratosthenes(M, N)
